### PR TITLE
Refactor python code to address Sonarcloud issues

### PR DESCRIPTION
**Pull request by rdmark**
- copy/move/delete file class methods now take Path objects as arguments
- The file download endpoint in the Web UI uses the safer download from dir method
- Simplified logging
- Merged nested if statements
- Removed naked handling of unknown error states
- Added fallback empty list for drive_properties, to avoid errors when json file is missing or broken
- Move drive_properties to env[]
- Constants for common error messages
- Dummy variable for list comprehension

### CODE REVIEW

File: python/web/tests/api/test_auth.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 # route("/login", methods=["POST"])
 def test_login_with_valid_credentials(pytestconfig, http_client_unauthenticated):
+    # Note: This test depends on the rascsi group existing and 'username' a member the group
```
Comment (nucleogenic) [inline python/web/tests/api/test_auth.py:6]: When it comes to testing, I think we have to make some assumptions about the state of the environment we are testing against, i.e. we care to know that the code works, and we test that against a given configuration; the objective isn't for the tests to work across many environmental permutations. For me that's going to be a standard RaSCSI image (on an RPi), or the Docker environment (on the dev's computer, or in CI). The comment is fine, just setting out my view on test environments!

File: python/common/src/rascsi/file_cmds.py [hunk 2]
```diff
@@ -305,10 +321,13 @@ def extract_image(self, file_path, members=None, move_properties_files_to_config
             properties_files_moved = []
             if move_properties_files_to_config:
                 for file in extract_result["extracted"]:
-                    if file.get("name").endswith(".properties"):
+                    logging.warning(file)
```
Comment (nucleogenic) [inline python/common/src/rascsi/file_cmds.py:324]: No log message, does this need removing?

File: python/common/src/rascsi/ractl_cmds.py [hunk 3]
```diff
@@ -260,18 +257,14 @@ def attach_device(self, scsi_id, **kwargs):
...
-            if "product" in kwargs.keys():
-                if kwargs["product"]:
-                    devices.product = kwargs["product"]
-            if "revision" in kwargs.keys():
-                if kwargs["revision"]:
-                    devices.revision = kwargs["revision"]
-            if "block_size" in kwargs.keys():
-                if kwargs["block_size"]:
-                    devices.block_size = int(kwargs["block_size"])
+            if "vendor" in kwargs and kwargs["vendor"]:
```
Comment (nucleogenic) [inline python/common/src/rascsi/ractl_cmds.py:260]: I haven't inspected the data, but I wonder if `kwargs.get("vendor")` could work here?

File: python/common/src/rascsi/sys_cmds.py [hunk 4]
```diff
@@ -32,8 +33,7 @@ def running_env():
                 .strip()
             )
         except subprocess.CalledProcessError as error:
-            logging.warning("Executed shell command: %s", " ".join(error.cmd))
-            logging.warning("Got error: %s", error.stderr.decode("utf-8"))
+            logging.warning(ERROR_FORMAT, error.cmd, error.stderr.decode("utf-8"))
```
Comment (nucleogenic) [inline python/common/src/rascsi/sys_cmds.py:36]: Agree it's better to have one log line, but I think `ERROR_FORMAT, x, y` is harder to grok. Maybe something like `PROCESS_ERROR, x, y` would abate that.

File: python/common/src/rascsi/sys_cmds.py [hunk 5]
```diff
@@ -8,6 +8,7 @@
 from re import findall, match
 from socket import socket, gethostname, AF_INET, SOCK_DGRAM
 
+ERROR_FORMAT = "[%s] %s"
```
Comment (nucleogenic) [inline python/common/src/rascsi/sys_cmds.py:11]: Will this error make sense in the logs?

File: python/common/src/rascsi/file_cmds.py [hunk 6]
```diff
@@ -258,11 +270,15 @@ def rename_file(self, file_path, target_path):
             }
 
     # noinspection PyMethodMayBeStatic
-    def copy_file(self, file_path, target_path):
+    def copy_file(self, file_dir, file_name, target_dir, target_name):
```
Comment (nucleogenic) [inline python/common/src/rascsi/file_cmds.py:273]: Since the strings are just concatenated in the method, what benefit do we get from accepting them separately in the new method signature? IMO it would be nicer to use `Path` objects, since most of the time we have a path we need to perform some kind of operation or check against it.

**Review by nucleogenic — APPROVED** [review]

File: python/web/tests/api/test_auth.py (hunk 1 above)
Comment (rdmark) [reply python/web/tests/api/test_auth.py:6]: This is the conclusion that I reached while pondering this earlier today, too. I ran into this failure when testing the authentication not active scenario, and thought having a comment here would be helpful for a future user who ran into the same failure and might have suspected the test was broken.

File: python/common/src/rascsi/sys_cmds.py (hunk 5 above)
Comment (rdmark) [reply python/common/src/rascsi/sys_cmds.py:11]: It looks like this:

```
2022-10-08 16:42:42,017 WARNING sys_cmds.py:50 [uname -3] uname: invalid option -- '3'
```

Let me iterate on a few variants and see if I find a better one.

File: python/common/src/rascsi/file_cmds.py (hunk 6 above)
Comment (rdmark) [reply python/common/src/rascsi/file_cmds.py:273]: I like the Path object approach. 
The trigger for the above change was Sonar complaining about user controller file paths. I may have gone overboard trying to address that. Using Path object validation is a much better way to secure the application, I think.

File: python/common/src/rascsi/sys_cmds.py (hunk 4 above)
Comment (rdmark) [reply python/common/src/rascsi/sys_cmds.py:36]: Sure, I'll tweak this.

File: python/common/src/rascsi/ractl_cmds.py (hunk 3 above)
Comment (rdmark) [reply python/common/src/rascsi/ractl_cmds.py:260]: I have a vague memory of running into some corner case errors with get() last year. But now when I test it all current workflows in the Web UI work fine. Let's try it!

Perhaps the one error state I can think of would be if kwargs is not a dict. Do you think we should add additional protection for this?

File: python/common/src/rascsi/sys_cmds.py (hunk 5 above)
Comment (rdmark) [reply python/common/src/rascsi/sys_cmds.py:11]: Updated!

File: python/common/src/rascsi/file_cmds.py (hunk 2 above)
Comment (rdmark) [reply python/common/src/rascsi/file_cmds.py:324]: Whoops, that was an artifact of my debugging.

File: python/common/src/rascsi/ractl_cmds.py (hunk 3 above)
Comment (nucleogenic) [reply python/common/src/rascsi/ractl_cmds.py:260]: Should be fine to assume kwargs is always a dict given the method signature 🤞

File: python/common/src/rascsi/file_cmds.py [hunk 7]
```diff
@@ -223,18 +223,17 @@ def copy_image(self, file_name, new_file_name):
...
         """
-        file_path = f"{file_dir}/{file_name}"
         parameters = {
             "file_path": file_path
         }
 
-        if os.path.exists(file_path):
-            os.remove(file_path)
+        if file_path.exists():
+            os.remove(file_path.as_posix())
```
Comment (nucleogenic) [inline python/common/src/rascsi/file_cmds.py:236]: Does `file_path.unlink()` work here?

File: python/common/src/rascsi/file_cmds.py [hunk 8]
```diff
@@ -247,20 +246,18 @@ def delete_file(self, file_dir, file_name):
...
         """
-        file_path = f"{file_dir}/{file_name}"
-        target_path = f"{target_dir}/{target_name}"
         parameters = {
             "target_path": target_path
         }
-        if os.path.exists(PurePath(target_path).parent):
-            os.rename(file_path, target_path)
+        if Path(target_path.parent).exists:
+            os.rename(file_path.as_posix(), target_path.as_posix())
```
Comment (nucleogenic) [inline python/common/src/rascsi/file_cmds.py:260]: Does `path.rename(target_path)` work here?

File: python/common/src/rascsi/file_cmds.py [hunk 9]
```diff
@@ -247,20 +246,18 @@ def delete_file(self, file_dir, file_name):
...
         Returns (dict) with (bool) status and (str) msg
         """
-        file_path = f"{file_dir}/{file_name}"
-        target_path = f"{target_dir}/{target_name}"
         parameters = {
             "target_path": target_path
         }
-        if os.path.exists(PurePath(target_path).parent):
-            os.rename(file_path, target_path)
+        if Path(target_path.parent).exists:
```
Comment (nucleogenic) [inline python/common/src/rascsi/file_cmds.py:259]: I think `target_path.parent.exists` would work here, as `parent` should be a Path-like object.

File: python/common/src/rascsi/file_cmds.py [hunk 10]
```diff
@@ -260,14 +272,16 @@ def rename_file(self, file_path, target_path):
...
+         - (Path) target_path for the name to copy to
         Returns (dict) with (bool) status and (str) msg
         """
         parameters = {
             "target_path": target_path
         }
-        if os.path.exists(PurePath(target_path).parent):
-            copyfile(file_path, target_path)
+        if Path(target_path.parent).exists:
+            copyfile(file_path.as_posix(), target_path.as_posix())
```
Comment (nucleogenic) [inline python/common/src/rascsi/file_cmds.py:284]: Are the calls to `as_posix()` to support Windows?

File: python/common/src/rascsi/file_cmds.py (hunk 7 above)
Comment (rdmark) [reply python/common/src/rascsi/file_cmds.py:236]: I can try it!

File: python/common/src/rascsi/file_cmds.py (hunk 8 above)
Comment (rdmark) [reply python/common/src/rascsi/file_cmds.py:260]: I didn't realize Path had these file operations. Note to self to read the documentation more closely.

File: python/common/src/rascsi/file_cmds.py (hunk 9 above)
Comment (rdmark) [reply python/common/src/rascsi/file_cmds.py:259]: Let me try it. I've had mixed success with this kind of syntax in the past, but it does definitely look cleaner.

File: python/common/src/rascsi/file_cmds.py (hunk 10 above)
Comment (rdmark) [reply python/common/src/rascsi/file_cmds.py:284]: No, it was the one method that I found in the docs that gives you the str representation. Do you know of a better way to achieve the same thing?

Comment (rdmark) [reply python/common/src/rascsi/file_cmds.py:284]: I should mention that I know that str() will get you the string representation, but I felt that using a class method would be the somewhat safer chocie!

Comment (nucleogenic) [reply python/common/src/rascsi/file_cmds.py:284]: If I understand correctly, when using `str(path)`, pathlib's internal `__str__` method will be called, which returns the path as a string.

Comment (rdmark) [reply python/common/src/rascsi/file_cmds.py]: That rings a bell. Alright, let's do this the straight-forward way.
